api_app/script_analyzers/observable_analyzers/urlscan.py

```diff
@@ -0,0 +1,84 @@
+import requests
+import time
+from api_app.exceptions import AnalyzerRunException
+from api_app.script_analyzers import classes
+from intel_owl import secrets
+
+
+class UrlScan(classes.ObservableAnalyzer):
+    base_url: str = "https://urlscan.io/"
+
+    def set_config(self, additional_config_params):
+        self.analysis_type = additional_config_params.get("urlscan_analysis", "search")
+        api_key_name = additional_config_params.get("api_key_name", "URLSCAN_API_KEY")
+        self.visibility = additional_config_params.get("visibility", "public")
+        self.__api_key = secrets.get_secret(api_key_name)
+
+    def __submit_query(self):
+        data = {"url": self.observable_name, "visibility": self.visibility}
+        uri = "api/v1/scan/"
+        try:
+            session = requests.Session()
```

Comment:
don't create a session here. Infact, create one in `run` function with the required `headers` and access it everywhere